tests/unit/recommenders/models/test_sar_singlenode.py

```diff
@@ -389,3 +390,25 @@ def test_get_normalized_scores(header):
     assert actual.shape == (2, 7)
     assert isinstance(actual, np.ndarray)
     assert np.isclose(expected, np.asarray(actual)).all()
+
+
+def test_match_similarity_type(header):
+    # store parameters in json
+    params_str = json.dumps({'similarity_type': 'jaccard'})
+    # load parameters in json
+    params = json.loads(params_str)
+
+    params.update(header)
+
+    model = SARSingleNode(**params)
+
+    train = pd.DataFrame(
+        {
+            header["col_user"]: [1, 1, 1, 1, 2, 2, 2, 2],
+            header["col_item"]: [1, 2, 3, 4, 1, 5, 6, 7],
+            header["col_rating"]: [3.0, 4.0, 5.0, 4.0, 3.0, 2.0, 1.0, 5.0],
+            header["col_timestamp"]: [1, 20, 30, 400, 50, 60, 70, 800],
+        }
+    )
+
+    model.fit(train)
```

Comment:
thanks for the input @danb27, not sure I understand what you are trying to check with this test. We already have a test that checks the fit function: https://github.com/microsoft/recommenders/blob/staging/tests/unit/recommenders/models/test_sar_singlenode.py#L92
There is another test that checks the init: https://github.com/microsoft/recommenders/blob/staging/tests/unit/recommenders/models/test_sar_singlenode.py#L74, however, it only checks it with the jaccard option. Maybe a good addition would be to add checks around lift and concurrence.

What do you think?

Reply:
@miguelgfierro This test isn't really testing the init or the fit method (sort of - it is really just testing 'is' vs. == in the fit function). This test would fail currently because the string comparison is done with 'is' instead of ==. Currently, if I do `SARSingleNode(similarity_type='lift', **other_params)` I will be able to fit successfully. BUT if I load in the similarity type from a JSON file, the fit method will give this error: 

https://github.com/microsoft/recommenders/blob/staging/recommenders/models/sar/sar_singlenode.py#L287 

even though I have spelled lift/jaccard/cooccurrence correctly. 

The purpose of this test is to ensure that this error is not raised regardless of whether the similarity type is provided directly in your code or if you are loading the desired similarity type from a json file (assuming you selected a valid similarity type)

Reply:
I changed the name of the test, added a comment to help clarify the purpose, and changed the similarity type used to test the functionality to 'lift' instead of 'jaccard' (just to test something other than jaccard in this one). 

Does that help clarify?

Reply:
this is fantastic Dan, I'll approve the PR